addons/test_sale_product_configurators/tests/test_sale_product_configurator.py

```diff
@@ -0,0 +1,214 @@
+            'display_type': 'radio',
+            'create_variant': 'dynamic'
+        }, {
+            'name': 'PA4',
+            'display_type': 'select',
+            'create_variant': 'no_variant'
+        }, {
+            'name': 'PA5',
+            'display_type': 'select',
+            'create_variant': 'no_variant'
+        }, {
+            'name': 'PA7',
+            'display_type': 'color',
+            'create_variant': 'no_variant'
+        }, {
+            'name': 'PA8',
+            'display_type': 'radio',
+            'create_variant': 'no_variant'
+        }])
+
+        self.env['product.attribute.value'].sudo().create([{
+            'name': 'PAV' + str(i),
+            'is_custom': i == 9,
+            'attribute_id': product_attribute.id
+        } for i in range(1, 11) for product_attribute in product_attributes])
+
+        product_template = self.product_product_custo_desk
+
+        self.env['product.template.attribute.line'].sudo().create([{
+            'attribute_id': product_attribute.id,
+            'product_tmpl_id': product_template.id,
+            'value_ids': [(6, 0, product_attribute.value_ids.ids)],
+        } for product_attribute in product_attributes])
+
+        self.assertEqual(len(product_template.product_variant_ids), 0)
+        self.assertEqual(
+            len(product_template.product_variant_ids.product_template_attribute_value_ids), 0,
+        )
+
+        self.start_tour(
+            "/web", 'sale_product_configurator_advanced_tour', login=self.env.user.login
+        )
+
+        # Ensures some dynamic create variants have been created by the configurator
+        self.assertEqual(len(product_template.product_variant_ids), 2)
+        self.assertEqual(
+            len(product_template.product_variant_ids.product_template_attribute_value_ids),
+            8,
+            "2 variants are created during the tour, with each 5 PAV, but only 8 distinct PAV."
+        )
+
+    @users('salesman', 'salemanager', 'admin')
+    def test_03_product_configurator_edition(self):
+        self.start_tour(
+            "/web", 'sale_product_configurator_edition_tour', login=self.env.user.login
+        )
+
+    @users('salesman')
+    def test_04_product_configurator_single_custom_value_for_salesman(self):
+        self._test_04_product_configurator_single_custom_value()
+
+    @users('salemanager')
+    def test_04_product_configurator_single_custom_value_for_manager(self):
+        self._test_04_product_configurator_single_custom_value()
+
+    @users('admin')
+    def test_04_product_configurator_single_custom_value_admin(self):
+        self._test_04_product_configurator_single_custom_value()
+
+    def _test_04_product_configurator_single_custom_value(self):
+        # Testing the single custom value creation: we can't use the same environment for different
+        # users
+
+        # Prepare relevant test data
+        # This is not included in demo data to avoid useless noise
```

Comment:
"Specifying multiple users with the users decorator doesn't trigger a rollback between each user, which means part of the test fails/is not covered unless triggered through different methods, each with their own targeted user".

or sthg with this meaning (could still be improved) :D.